File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>>ans;
        map<vector<int> , vector<string>>mp;
        vector<int>count(26 , 0);
        for (int i = 0 ; i < strs.size(); i++)
        {
            for (int j = 0 ; j < strs[i].size(); j++)
            {
                char c = strs[i][j];
                count[c-'a']++;
            }
            mp[count].push_back(strs[i]);
            for (int j = 0 ; j < 26; j++)count[j] = 0;
        }
        for (auto x : mp)
        {
            vector<string>temp;
            for (auto z : x.second)
                temp.push_back(z);
            ans.push_back(temp);
        }
        
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> mp = new HashMap<>();
        
        for (String str : strs) {
            // Convert the string to a character array
            char[] chars = str.toCharArray();
            // Sort the character array
            Arrays.sort(chars);
            // Convert the character array back to a string
            String sortedString = new String(chars);
            
            // Check if the sorted string is already a key in the map
            if (!mp.containsKey(sortedString)) {
                mp.put(sortedString, new ArrayList<>());
            }
            mp.get(sortedString).add(str);
        }
        
        // Collect the grouped anagrams into a list
        List<List<String>> ans = new ArrayList<>(mp.values());
        
        return ans;
    }
}
